lib/devise_saml_authenticatable/routes.rb

```diff
@@ -1,12 +1,23 @@
 ActionDispatch::Routing::Mapper.class_eval do
   protected
   def devise_saml_authenticatable(mapping, controllers)
-    resource :session, :only => [], :controller => controllers[:saml_sessions], :path => "" do
-      get :new, :path => "saml/sign_in", :as => "new"
-      post :create, :path=>"saml/auth"
-      match :destroy, :path => mapping.path_names[:sign_out], :as => "destroy", :via => mapping.sign_out_via
-      get :metadata, :path=>"saml/metadata"
-      match :idp_sign_out, :path=>"saml/idp_sign_out", via: [:get, :post]
+    if ::Devise.route_helper_prefix
+      prefix = ::Devise.route_helper_prefix
+      resource :session, only: [], controller: controllers[:saml_sessions], path: '' do
+        get :new, path: 'saml/sign_in', as: "new_#{prefix}"
+        post :create, path: 'saml/auth', as: prefix
+        match :destroy, path: mapping.path_names[:sign_out], as: "destroy_#{prefix}", via: mapping.sign_out_via
+        get :metadata, path: 'saml/metadata'
+        match :idp_sign_out, path: 'saml/idp_sign_out', as: "idp_destroy_#{prefix}", via: %i[get post]
+      end
+    else
+      resource :session, only: [], controller: controllers[:saml_sessions], path: '' do
+        get :new, path: 'saml/sign_in', as: 'new'
+        post :create, path: 'saml/auth'
+        match :destroy, path: mapping.path_names[:sign_out], as: 'destroy', via: mapping.sign_out_via
+        get :metadata, path: 'saml/metadata'
+        match :idp_sign_out, path: 'saml/idp_sign_out', via: %i[get post]
+      end
```

Comment:
I'll take a look at adding routing specs for this before merging.

Reply:
I imagine the specs could look something like what I have locally but never committed. Here's a gist of the specs hopefully a quick copy & paste will have them working for you: https://gist.github.com/scottserok/6ab54f397319ba9d9e095926dd5352a6 
Thanks for reviewing my PR Adam!

Reply:
Great start! I had to mess around to get the parameter picked up by the context, since the routes need to be re-drawn, so I ended up using an Engine in the test file.